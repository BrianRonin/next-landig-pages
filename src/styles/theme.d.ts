// import 'styled-components'
// declare module 'styled-components' {
//   interface x {
//     colors: {
//       primaryColor: string
//       secondaryColor: string
//       mainBg: string
//       lightBg: string
//       secondaryBg: string
//       white: string
//       mediumGray: string
//       darkBg: string
//     }
//     fonts: {
//       family: {
//         default: string
//         secondary: string
//       }
//       sizes: {
//         xsmall: string
//         small: string
//         medium: string
//         large: string
//         xlarge: string
//         xxlarge: string
//         huge: string
//         xhuge: string
//       }
//     }
//     sizes: {
//       xsmall: string
//       small: string
//       medium: string
//       large: string
//       xlarge: string
//       xxlarge: string
//       huge: string
//       xhuge: string
//     }
//     media: {
//       medium: string
//     }
//     spacings: {
//       xsmall: string
//       small: string
//       medium: string
//       large: string
//       xlarge: string
//       xxlarge: string
//       huge: string
//       xhuge: string
//     }
//   }

//   export interface DefaultTheme extends x {}
// }
