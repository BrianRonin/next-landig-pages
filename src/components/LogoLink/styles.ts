import styled from 'styled-components'
export const Main = styled.a`
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;

  > img {
    max-height: 6rem;
  }
`
