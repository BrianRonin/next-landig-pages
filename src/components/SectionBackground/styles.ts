import styled, { css, DefaultTheme } from 'styled-components'

type background = {
  background: boolean
}

const modeBackground = (theme: DefaultTheme) => css`
  color: ${theme.colors.white};
  background: ${theme.colors.primaryColor};
`

export const Main = styled.div<background>`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && modeBackground(theme)}
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`
