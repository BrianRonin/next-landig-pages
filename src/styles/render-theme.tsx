import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from './theme'

type renderThemeProps = {
  children: React.ReactNode
}

export const renderTheme = ({ children }: renderThemeProps) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}
