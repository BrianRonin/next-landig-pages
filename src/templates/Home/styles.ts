import styled, { css } from 'styled-components'

export const container = styled.div`
  ${({ theme }) => {
    return css`
      background: ${theme.colors.secondaryBg};
    `
  }}
`
