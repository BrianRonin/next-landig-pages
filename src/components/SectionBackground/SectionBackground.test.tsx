import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/render-theme'
import { SectionBackground } from './index'

describe('<SectionBackground />', () => {
  it('should render with background dark', () => {
    const { container } = renderTheme(
      <SectionBackground sectionId='1' background={true}>
        <h1>teste</h1>
      </SectionBackground>,
    )
    expect(screen.getByRole('heading')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
  it('should render with background white', () => {
    const { container } = renderTheme(
      <SectionBackground sectionId='1' background={false}>
        <h1>teste</h1>
      </SectionBackground>,
    )
    expect(screen.getByRole('heading')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
