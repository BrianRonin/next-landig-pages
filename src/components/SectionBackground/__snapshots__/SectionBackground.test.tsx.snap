// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with background dark 1`] = `
.c0 {
  background: #FFFFFF;
  color: #0A1128;
  color: #FFFFFF;
  background: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
  position: relative;
}

<div>
  <div
    class="c0"
    id="1"
  >
    <section
      class="c1"
    >
      <h1>
        teste
      </h1>
    </section>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with background white 1`] = `
.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
  position: relative;
}

<div>
  <div
    class="c0"
    id="1"
  >
    <section
      class="c1"
    >
      <h1>
        teste
      </h1>
    </section>
  </div>
</div>
`;
